source/Tokens/Literals/Number.ts

```diff
@@ -0,0 +1,164 @@
+						case "2": value += 2; original += reader.read(); break
+						case "3": value += 3; original += reader.read(); break
+						case "4": value += 4; original += reader.read(); break
+						case "5": value += 5; original += reader.read(); break
+						case "6": value += 6; original += reader.read(); break
+						case "7": value += 7; original += reader.read(); break
+						case "8": value += 8; original += reader.read(); break
+						case "9": value += 9; original += reader.read(); break
+						default: result = new Number(divisor ? value / divisor : value, original, reader.mark()); continue
+					}
+					value *= 10
+					if (divisor)
+						divisor *= 10
+				}
+			} else {
+				result = null
+			}
+			return result
+		}
+		private static isNumber(character: string): boolean {
+			return character >= "0" && character <= "9"
+		}
+		private static scanOctal(reader : IO.BufferedReader): Token {
+			var result: Token
+			if (reader.peek(2) == "0o") {
+				var original = reader.read(2)
+				var value: number
+				var divisor: number
+				while (!result) {
+					switch (reader.peek()) {
+						case ".":
+							original += reader.read()
+							if (divisor)
+								result = new Number(divisor ? value / divisor : value, original, reader.mark())
+							else
+								divisor = 1
+							continue
+						case "_": original += reader.read(); continue
+						case "0": value += 0; original += reader.read(); break
+						case "1": value += 1; original += reader.read(); break
+						case "2": value += 2; original += reader.read(); break
+						case "3": value += 3; original += reader.read(); break
+						case "4": value += 4; original += reader.read(); break
+						case "5": value += 5; original += reader.read(); break
+						case "6": value += 6; original += reader.read(); break
+						case "7": value += 7; original += reader.read(); break
+						default: result = new Number(divisor ? value / divisor : value, original, reader.mark()); continue
+					}
+					value *= 8
+					if (divisor)
+						divisor *= 8
+				}
+			} else {
+				result = null
+			}
+			return result
+		}
+		private static scanBinary(reader : IO.BufferedReader): Token {
+			var result: Token
+			if (reader.peek(2) == "0b") {
+				var original = reader.read(2)
+				var value: number
+				var divisor: number
+				while (!result) {
+					switch (reader.peek()) {
+						case ".":
+							original += reader.read()
+							if (divisor)
+								result = new Number(divisor ? value / divisor : value, original, reader.mark())
+							else
+								divisor = 1
+							continue
+						case "_": original += reader.read(); continue
+						case "0": value += 0; original += reader.read(); break
+						case "1": value += 1; original += reader.read(); break
+						default: result = new Number(divisor ? value / divisor : value, original, reader.mark()); continue
+					}
+					value *= 2
+					if (divisor)
+						divisor *= 2
+				}
+			} else {
+				result = null
+			}
+			return result
+		}
+	}
+}
```

Comment:
1) It is something currently not supported by OOC but that i thought might make sense to support.
2) Of course but it will add more complication as will supporting _ which is a part of ooc.

In  the end I will focus on moving forward and not revisit this for now. In the end I tend to prefer a solution only going through the string representation once and not twice.